### ISSUE TITLE
Spectrogram's canvas does not full fill the container of it

### ISSUE THREAD

**Issue by KrazyPhish** [katspaugh/wavesurfer.js#3140]
## Bug description
I was doing a Vue2 to Vue3 project, found that spectrogram's canvas cannot full fill its parent element.
But it works well in my Vue2 project.
Infomation of the versions: 
 my Vue2 project: wavesurfer.js 6.2.0 (2022-05-16)
 my Vue3 project: wavesurfer.js latest

## Environment
 - Dev: VSCode, TypeScript, Vue3
 - Browser: Chrome - version 116.0.5845.97
 - OS: windows 11

## Minimal code snippet
In Vue2:
 plugins.push(
  spectrogram.create({
    container: this.$refs.waveSpectrogramRef,
    labels: true,
    height: height * this.verticalZoom.value,
    colorMap
  })
)

In Vue3: 
 const spectrogram: SpectrogramPlugin = SpectrogramPlugin.create({
  container: waveSpectrogramRef.value as HTMLElement,
  labels: true,
  height: height * state.verticalZoom,
  colorMap
})

The value height equals its parent's clientHeight, it's a constant value, so is the verticalZoom value.

## Expected result
Like wavesurfer works in my Vue2 project, spectrogram's canvas fully fill its parent element.

## Obtained result
Same as Expected result

## Screenshots
 The parent element div.
![image](https://github.com/katspaugh/wavesurfer.js/assets/52484425/da0e40df-a0bd-4587-8b21-9478153105ec)
 The left side label canvas, this means spectrogram tried to fill the element, I guess.
![image](https://github.com/katspaugh/wavesurfer.js/assets/52484425/f679577c-d8c1-474b-aa1f-a5cb08d660dc)
The style attributes show that the canvas has fully filled its parent element, there must be some problems with canvas while the plugin working with canvas.
![image](https://github.com/katspaugh/wavesurfer.js/assets/52484425/4fca79dc-2110-4eba-a8da-20549354d6d3)

**Comment by katspaugh** [comment]
You might need to increase the sampleRate for it to have the data to fill the chart. Try passing 22050 or 44100 like here: https://wavesurfer-js.org/examples/?spectrogram.js

**Comment by KrazyPhish** [comment]
Attribute sampleRate 22050 works the same as my last try, 44100 makes a little diffrence but still didn't solve the problem, here's the pic in 44100 samepleRate, it seems that only expanded the label values:
![image](https://github.com/katspaugh/wavesurfer.js/assets/52484425/4dad7936-eb37-404d-8c13-74938ee99397)

**Comment by katspaugh** [comment]
I see, thanks for checking this. Looks like a regression in v7, since you said it worked well in v6. 🐞

**Comment by KrazyPhish** [comment]
Yep, the version of  6.2.0 worked well, but the problem above was found while using the version of 7.1.5. Hope this can help.

**Comment by Richard-2-4-2** [comment]
hi all, is there the same isuue? 
![image](https://github.com/katspaugh/wavesurfer.js/assets/148053489/c06956ca-7905-434c-89a8-b6af30b9b43e)

**Comment by katspaugh** [comment]
Try decreasing your sample rate

![Screenshot 2023-10-16 at 11 20 22](https://github.com/katspaugh/wavesurfer.js/assets/381895/8f326aed-3428-454a-9a34-4035a229dfc1)

**Comment by Richard-2-4-2** [comment]
thanks for your answer. no changes (on left). on right, image of audacity for comparison. seems that frequencies beyond 4000hz are not drawn 
![image](https://github.com/katspaugh/wavesurfer.js/assets/148053489/62a93fa2-d523-4639-bf41-d0402226e14a)

**Comment by KrazyPhish** [comment]
It looks like decreasing the sampleRate value solved your problem in some way? I've tried it too, but makes no diffrence in my case

**Comment by katspaugh** [comment]
Try also adjusting the min and max frequency accordingly.

https://wavesurfer.xyz/docs/types/plugins_spectrogram.SpectrogramPluginOptions

> `frequencyMax?: number`
> Max frequency to scale spectrogram. Set this to samplerate/2 to draw whole range of spectrogram.

**Comment by Richard-2-4-2** [comment]
my sample is 44100 hz. i can give a value up to 11050hz or i must use these value ?

**Comment by Richard-2-4-2** [comment]
> It looks like decreasing the sampleRate value solved your problem in some way? I've tried it too, but makes no diffrence in my case
sorry, i copy only one layer of photoshop. the two graphes are for comparison to audacity ( i want to have similar appearence)

**Comment by katspaugh** [comment]
You can set any value down to 8000 for sampleRate. In Chrome down to 3000.

**Comment by Richard-2-4-2** [comment]
for spectrogram : 
if i define height: 500 : 181 pixels drawn
height: 2000 : 185 pixels drawn
is it normal?
i dont understand the samplerate option.  the value is given by the file, non ?
something special to chrome compared to others?

**Comment by KrazyPhish** [comment]
![image](https://github.com/katspaugh/wavesurfer.js/assets/52484425/462cddce-ec66-42a1-9bdf-6c6041179218)
I am very glad to see this to solve my problem, temporarily. I guess that setting the sampleRate and frequencyMax may cut the canvas, however, didn't like the docs said to set the frequencyMax to sampleRate/2 but the sampleRate/4.

**Comment by Richard-2-4-2** [comment]
[KrazyPhish](https://github.com/KrazyPhish), how is the samplerate of your audio file? wich values are you using for sampleRate and frequencyMax ?

**Comment by KrazyPhish** [comment]
I am not sure about my files sample rate range cause the file was somehow just use to test, 22050 for the sampleRate and 5512.5 for the frequencyMax are my settings.

**Comment by Richard-2-4-2** [comment]
my sample is Mono, 44100Hz, 32bits and i would have the whole spectrum as audacity . previous version i used worked.
 
![image](https://github.com/katspaugh/wavesurfer.js/assets/148053489/f0a4914c-2f68-4845-8e80-a55b4e8604d2)

**Comment by katspaugh** [comment]
I think this is now fixed in the latest version, can anyone confirm?

**Comment by KrazyPhish** [comment]
I've tried the version of 7.8.15, it worked, I raised the sampleRate value, haven't seen the problem anymore.
